### ISSUE TITLE
ERRINFO_DECRYPT_FAILED (0x00001192) and Connection reset by peer

### ISSUE THREAD

**Issue by bakytn** [FreeRDP/FreeRDP#433]
When I try to work with redirected disk it's disconnecting with the following error:

user@host: /# xfreerdp -f -u jedi -p knight -x m --ignore-certificate --no-tls --no-nla --plugin rdpdr --data disk:FLASH:/mnt/ -- 10.0.1.76

I tried to work without --no-tls and --no-nla settings
user@host: /# xfreerdp -f -u jedi -p knight -x m --ignore-certificate --plugin rdpdr --data disk:FLASH:/mnt/ -- 10.0.1.76

and
user@host: /# xfreerdp -f -u jedi -p knight -x m  --plugin rdpdr --data disk:FLASH:/mnt/ -- 10.0.1.76

and
user@host: /# xfreerdp -f -u jedi -p knight -x m --no-rdp  --plugin rdpdr --data disk:FLASH:/mnt/ -- 10.0.1.76

the same issue:

loading plugin rdpdr
Warning xf_GetWindowProperty (140): Property 340 does not exist
connected to 10.0.1.76:3389
registered device #1: FLASH (type=8 id=1)
Warning disk_process_irp (538): MajorFunction 0x11 not supported
Warning disk_process_irp (538): MajorFunction 0x11 not supported
Warning disk_process_irp (538): MajorFunction 0x11 not supported
Warning disk_process_irp (538): MajorFunction 0x11 not supported
ERRINFO_DECRYPT_FAILED (0x00001192):
(a) Decryption using Standard RDP Security mechanisms (section 5.3.6) failed.
(b) Session key creation using Standard RDP Security mechanisms (section 5.3.5) failed.
recv: Connection reset by peer
Failed to check FreeRDP file descriptor

In the Event Logs of the server

I am getting: Event ID: 50 Source: TermDD

"The RDP protocol component "DATA ENCRYPTION" detected an error in the protocol stream and has disconnected the client."

and sometimes "The RDP protocol component X.224 detected an error in the protocol stream and has disconnected the client."

(the same Event ID and Source just different messages)

**Comment by lemonmojo** [comment]
I also have several customers who reported this issue to me. Mostly with WinXP and Win2k3 machines. This is on current master.

**Comment by Rexilion** [comment]
We are having disconnects too with w2k3 machines, not w2k8.

[gebruiker@TPC-12-03 ~]$ xfreerdp /v:192.168.0.26 /f
[10:45:09:984] [738:739] [ERROR][com.freerdp.legacy] - ERRINFO_DECRYPT_FAILED (0x00001192):
(a) Decryption using Standard RDP Security mechanisms (section 5.3.6) failed.
(b) Session key creation using Standard RDP Security mechanisms (section 5.3.5) failed.

[10:45:09:985] [738:739] [ERROR][com.freerdp.legacy] - Failed to check FreeRDP file descriptor

Should I open a separate bug for this?

**Comment by lemonmojo** [comment]
@Rexilion I think noone in this thread mentioned issues with Win2k8. Looks like the problem is isolated to older Windows Versions (Win2k, Win2k3 and WinXP).

**Comment by Rexilion** [comment]
@lemonmojo, ok so a separate bug it is then?

rdesktop does not have this issue. We tested that today.

**Comment by lemonmojo** [comment]
@Rexilion I think it's not necessary to create a new issue because you're, like everyone else am having these issues with older Windows versions, right?

**Comment by AurimasF** [comment]
We have the same problem with Win2k3 and FreeRDP master.
Is it possible to somehow capture FreeRDP RDP communication and see what exactly causes this issue?

**Comment by Rexilion** [comment]
@lemonmojo The initial issue mentions redirected disks. Not the disconnect on w2k3 machines.

**Comment by lemonmojo** [comment]
@Rexilion Yes but I think this is just a coincidence... Anyway, that's up for @bakytn to answer.

**Comment by bmiklautz** [comment]
@lemonmojo do you still have reports about this issue? I couldn't reproduce it with any means ;(.

**Comment by lemonmojo** [comment]
@bmiklautz Yes, it's a rare issue but the last report came in last July. I was also never able to reproduce this on purpose. Of course it always only occurred when I had no chance of debugging...

**Comment by bmiklautz** [comment]
@lemonmojo thank you for the feedback. So I'll leave the issue open. Hope we get it fixes at some point ;).

**Comment by cloudecker** [comment]
We are having disconnects too with win7 machines，mac client .
 ERRINFO_DECRYPT_FAILED (0x00001192):
(a) Decryption using Standard RDP Security mechanisms (section 5.3.6) failed.
(b) Session key creation using Standard RDP Security mechanisms (section 5.3.5) failed.
[ERROR][com.freerdp.core] - freerdp_check_fds() failed - 0
then disconnected.

**Comment by AurimasF** [comment]
Exactly the same error as @cloudecker and disconnect (3 in the last 10 minutes). One of these disconnects was registered in Windows 2003 Event log: Source: TermDD; Event ID: 50
The RDP protocol component "DATA ENCRYPTION" detected an error in the protocol stream and has disconnected the client.

**Comment by micsutka** [comment]
Exactly the same error here.
Client: xubuntu 14.04 and freerdp 2.0.0~dev~git20160331+dfsg-0trusty1
Server: WinXP Pro SP3 32bit
The server os (winxp) is a virtual machine installed on qemu-kvm 2.0.0+dfsg-2ubuntu1.22
The both server and the client are on the same hardware (amd phenom II x4 965)

command line:
xfreerdp /v:192.168.122.102 +compression /u:john /p:doe +clipboard /size:1864x1080 /t:windowtitle /sound /gdi:sw &

The error message is the exact same as @cloudecker has mentioned.
The windows event log contains the exact same as @AurimasF has mentioned.

**Comment by amrheing** [comment]
Hi all,
i got the same error. But i find out this:

When copying in ONE Explorer Window from the mounted drive to an Folder on D: then it crashes.

When copying from an Explorer Window from the mounted drive to the Desktop - it works.

May be it is the behaviour, then copying between 2 explorer Windows.

greetings, Gerald

**Comment by sailingbikeruk** [comment]
I am afraid it appears that it is still happening, I use Royal TS on MAC OS 10.12.5. Royal TS is a multi protocol multi connection GUI that uses FreeRDP to provide RDP connections.

I am seeing this issue when connecting to any server with Server 2K3 or earlier (yes there are still a lot of them about).

I have no definitive evidence  but it appears to get more frequent the longer the connection is open.. hearsay maybe just the way it feels.

![screenshot 2017-06-03 08 57 09](https://cloud.githubusercontent.com/assets/29166333/26756229/661c2b40-4895-11e7-9e0f-f8219b1ddf3c.png)

**Comment by antenore** [comment]
@sailingbikeruk please provide the exact version of both Remmina and freerdp

**Comment by antenore** [comment]
Sorry mixed up the Remmina and FreeRDP thread.

@sailingbikeruk

**Comment by nanxiongchao** [comment]
Here I describe my test:
I found that in the process of shaking the mouse if the insertion of other events will delay the crash, such as key-press-event or focus-in-event, the most rapid crash method to reproduce is in the remote desktop continuous shaking the mouse without any other operations, probably Spend 30 seconds each time.
So the reason why this problem has not been frequently discovered is that if other events are operated, causing the crash problem will result in an unknown time delay.
I also tested the previous versions, found from the freerdp-1.0.1 version to the current latest has been the problem.
View server logs:
<pre>
TermDD
RDP protocol component "DATA ENCRYPTION" found an error in the protocol flow and interrupted the client connection
</pre>

Also, from https://github.com/FreeRDP/Remmina/issues/1270
weberhofer said:
"Just an idea: Support for Windows Server 2003 ended 2015. It can be that SSL on current distributions does no longer support some SSL features (ciphers or whatever) which for security reasons must no longer be used."
